src/python/pants/engine/native.py

```diff
@@ -787,6 +789,10 @@ def buffer(self, cdata):
   def to_ids_buf(self, types):
     return self.context.type_ids_buf([TypeId(self.context.to_id(t)) for t in types])
 
+  def decompress_tarball(self, tarfile_path, dest_dir):
+    result = self.lib.decompress_tarball(tarfile_path, dest_dir)
+    self.context.raise_or_return(result)
```

Comment:
You'll need the `return`, still :)

Reply:
Will do.

In this particular case, it is just a `Ok(())` coming from rust, so I can keep the caller as is, right? as in not having to handle the result, but only catching the exception.
```
  def extract(self):
    # Note(yic): unlike the python implementation before, now we do not update self._relpath
    # after the extraction.
    try:
      self.NATIVE_BINARY.decompress_tarball(self._tarfile.encode('utf-8'),
                                                   self._artifact_root.encode('utf-8'))
    except Exception as e:
      raise ArtifactError("Extracting artifact failed:\n{}".format(e))
```

Reply:
Yeah, in this case it doesn't actually make a difference, but when someone changes the rust to return something, it will be totally non-obvious to them that the python bridge code is just swallowing the value - I've made this mistake a bunch of times, and it's super frustrating :)